### PR TITLE
Compatibility with python 3.12

### PR DESCRIPTION
**Pull request by CyberDem0n**
- monkey patch `jsonlogger.RESERVED_ATTRS` to hide new attribute in `LogRecord`
- "silence" warning about `atetime.datetime.utcnow()`
- run some tests with python 3.12
- bump actions versions to silence complains about Node version
- fix PATH to Postgres binaries on MacOS

### CODE REVIEW

File: patroni/log.py [hunk 1]
```diff
@@ -350,6 +350,9 @@ def _get_json_formatter(self, logformat: type_logformat, dateformat: Optional[st
 
         try:
             from pythonjsonlogger import jsonlogger
+            if hasattr(jsonlogger, 'RESERVED_ATTRS') and 'taskName' not in jsonlogger.RESERVED_ATTRS:
+                # compatibility with python 3.12, that added a new attribute to LogRecord
+                jsonlogger.RESERVED_ATTRS = jsonlogger.RESERVED_ATTRS + ('taskName',)
```
Comment (hughcapet) [inline patroni/log.py:355]: am i missing smth and you don't use `+=` intentionally?

**Comment by hughcapet** [comment]
👍

**Comment by hughcapet** [comment]
https://github.com/zalando/patroni/blob/025098c26f691ca328967eec53cc285e0d6ecef5/.github/workflows/release.yaml#L17

and maybe also take advantage of this PR to update version here:
https://github.com/zalando/patroni/blob/025098c26f691ca328967eec53cc285e0d6ecef5/.github/workflows/release.yaml#L42

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**